lib/auth0/api/v2/resource_servers.rb

```diff
@@ -62,6 +62,17 @@ def delete_resource_server(resource_server_id)
           delete(path)
         end
 
+        # Updates a resource server.
+        # @see https://auth0.com/docs/api/management/v2#!/Resource_Servers/patch_resource_servers_by_id
+        # @param id [string] The id or audience of the resource server to update.
+        # @param options [hash] The Hash options used to define the resource servers's properties.
+        def patch_resource_server(id, options)
+          raise Auth0::InvalidParameter, 'Must specify a resource server id' if id.to_s.empty?
```

Comment:
Let's say:

> Must specify a resource server ID or audience

... since it can accept both.

Reply:
Actually, on this topic, should we URI-escape this parameter, or except the caller to do it? Obviously irrelevant if the caller is passing an id, but if passing an audience, and the audience is of the common form `https://something.com/` then that needs to be URI-escaped before being added to the PATH string.

In my PR I left it up the to caller to URI escape audiences, since the `delete_resource_server`  method (the only existing method that puts an id-or-audience in the URI's path segment) doesn't handle URI escaping automatically, but an argument could be made to add it. Do you want me to go ahead and add that here, and to the delete method as well? Or, since that's theoretically a backwards-incompatible change (anyone previously using the delete method with an audience would be escaping before passing it in, and if we start escaping as well then it ends up double-escaped), should it wait for  a separate PR that could be part of 5.0.0?

Reply:
That's a great question. 

I would prefer it be done in the SDK but none of the other methods I looked at for any of the endpoints do so I think we should be consistent here. We're putting together a roadmap for a major release on this library and I'll bring that up for discussion.

Reply:
It's worth noting that most endpoints don't accept anything in the the path segment of the URL that could ever possibly need URI escaping; the resource servers endpoint is the only one I can think of off the top of my head.

Just to play devils advocate, you could argue that adding this functionality isn't a compatibility-breaking change because the library did not previously document any support for accepting an audience rather than an id in these methods, so anyone who was passing one in was relying on undocumented behavior (of the library; the actual REST API does document this of course). That said, I'm not sure I buy that argument myself and am content to wait for 5.0.0 to introduce the automatic escaping.

Reply:
It's definitely the minority of cases, true. But also a consideration for URL parameters.

Every time I talk myself into changing something that shouldn't be a breaking change but is, there is inevitably someone affected! I think it makes more sense to address it gem-wide in a major. I appreciate the opportunity to talk it through, though.